cli/popper/commands/cmd_run.py

```diff
@@ -96,6 +96,39 @@ def cli(ctx, pipeline, timeout, skip, ignore_errors, output,
         pu.fail("Failed to execute pipeline")
 
 
+def check_for_env_vars(args):
```

Comment:
We can avoid having this function if we create a list of executions. So first get a list of executions for a pipeline, and then we pass the list to the `run_pipeline` function. For pipelines where there's no `env_vars` defined, we create one execution (i.e. the empty set of vars).